src/valuegraph.jl

```diff
@@ -628,7 +628,57 @@ end
 #  rem_vertex!
 #  ------------------------------------------------------
 
-# TODO
+function LG.rem_vertex!(g::ValGraph, v::Integer)
+    # Procedure: swap v and n, then remove n
+    v in vertices(g) || return false
+    n = nv(g)
+    self_loop_n = false  # true if n is self-looped (see #820)
+
+    # Swap the vertex values of v and n
+    if v != n
+        for i in 1:length(g.vertexvals)
+            g.vertexvals[i][v] = g.vertexvals[i][n]
+        end
+    end
+
+    # Swap the edges and edge values of v and n
+
+    # remove the in_edges incident to v
+    neigs_v = copy(inneighbors(g, v))
+    for s in neigs_v
+        rem_edge!(g, s, v)
+    end
+    values_selfloop = nothing # TODO: fix this type instability
```

Comment:
I would not worry too much about type instability. If you have some variable of type `Union{T, Nothing}` for some type `T`, then Julia will still be able to create efficient code. In general, Julia can generate efficient code, as long as there are not too many types involved in an `Union`. That's why using `@code_warntype` will also color this variable yellow and not red.